### DISCUSSION TITLE
Linking Laragon www and data to network drive

### DISCUSSION THREAD

**Discussion by BackToBasiks** [leokhoa/laragon#659] [Q&A]
Curious if anyone has had luck with this. We're trying to make it so the document root and data directories are pulled from a network drive. It works with the www file, but when trying to connect to the database via phpmyadmin we get
`mysqli::real_connect(): (HY000/2002): No connection could be made because the target machine actively refused it.`

If we switch the data directory to our machine running Laragon it pulls up the database no problems. The data folder on our machine and network drive is the same, we just copied it from the network to our machine.

Edit: Weirdly enough after we switched the data folder back to our network drive it started working after Laragon was running for a couple of minutes? Any ideas on this?

Another Edit: While it seemed to work after a couple minutes, it is very slow. :( 
Open to ideas on how to make it faster. Speeds from the network drive are 100MBPS so it's not slow in that regards, just takes forever for the next page to load.

**Comment by Pen-y-Fan** [comment]
My laptop has a Samsung SSD drive, the utility Samsung Magician has it running as 3,217 MB/s read and 3,202 MB/s write. 

A network drive is always going to be noticeably slower, an operation that takes a second could take minutes.

**Comment by BackToBasiks** [comment]
I ended up building a server on a mini pc and setting up docker xampp with a bind to a network drive for www. Really makes you miss Laragon cause the Virtual Hosts is so much easier with it. Loading files to edit from the network drive is still a bit slow, but no where near as slow as loading an entire page with everything on a network drive.

My main concern was with syncing the database. Since git can easily handle syncing files, I'll probably repurpose the server to run mysql and use git to sync files across multiple devices so that I can continue to use Laragon.

**Comment by pravnkay** [comment]
You can try using Syncthing (Synctrazorm for GUI).
I am using this to sync my folders between my work PC and home PC. So, whatever I work during the evenings and weekends in my home PC will be synced when I turn my work PC on. The catch is that your home PC needs to running for this to work. Or you can setup a common PC which will be always turned on like a server, that is only going to share files to Home & Work PC and not going to be used for anything else. A pi or an old PC can do this, I think
